lib/quantization/src/encoded_storage.rs

```diff
@@ -116,10 +131,24 @@ impl EncodedStorageBuilder for TestEncodedStorageBuilder {
     type Storage = TestEncodedStorage;
 
     fn build(self) -> std::io::Result<Self::Storage> {
-        Ok(TestEncodedStorage { data: self.data })
+        Ok(TestEncodedStorage {
+            data: self.data,
+            quantized_vector_size: self.quantized_vector_size.unwrap_or_default(),
+        })
     }
 
     fn push_vector_data(&mut self, other: &[u8]) {
+        debug_assert_ne!(other.len(), 0, "Cannot push empty vector data");
+        match self.quantized_vector_size {
+            Some(size) => {
+                if other.len() != size {
+                    panic!("Data length is not equal to quantized vector size");
+                }
+            }
+            None => {
+                self.quantized_vector_size = Some(other.len());
+            }
+        }
```

Comment:
Can't we somehow enforce to specify the quantized size when creating this structure, so we don't have to derive it now?

Reply:
I agree with you. Can I do it as a separate PR? Your proposal causes changes in each quantization test.